common/quotas/collection.go

```diff
@@ -29,6 +29,12 @@ type Collection struct {
 	limiters map[string]Limiter
 }
 
+type ICollection interface {
+	For(key string) Limiter
+}
+
+var _ ICollection = (*Collection)(nil)
```

Comment:
`*Collection` only exposes this one API.  surprisingly we depend on `*Collection` everywhere rather than an interface.

this whole quotas/*.go set of code badly needs a cleanup pass, especially now that we have `clock.Ratelimiter`.  but that's not part of this PR.

Reply:
urgh, yeah, I think by by convention a lot of us (or at least me) expected the primary noun of a package (ie Collection) to be the interface and the impl to be hidden. A lot of this is probably inertia

Reply:
yea, generally I think we should follow that naming pattern (though private impl).  just didn't seem worth imposing it in this PR 👍